From 92b4a700078e439754310fb6745868bdd488eea3 Mon Sep 17 00:00:00 2001
Message-Id: <92b4a700078e439754310fb6745868bdd488eea3.1425610844.git.chang-joon.lee@intel.com>
In-Reply-To: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
References: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Mon, 9 Feb 2015 18:22:50 +0530
Subject: [PATCH 14/27] SQUASHME! [VPG]: video/adf/intel: Fix screen shift on
 MIPI for dual displays suspend

In case of dual displays HDMI+MIPI, there was a sporadic shift seen on MIPI after resume,
due to DPI FIFO underrun, fixing the issue by giving delay for the FIFO to be empty.

SQUASHME! - This patch should be squashed into the following existing patch:
	Author: Gaurav K Singh <gaurav.k.singh@intel.com>
	Date:   Mon Feb 2 14:53:38 2015 +0530
	Change-Id: I326890542c453d01c5ef0a1781407dfe6352b3b3
	MUST_REBASE [VPG]: drivers/video/adf: Fix sporadic screen shift on DSI

Issue: GMINL-5611
Change-Id: I80a77f9ce0a0281e4d457329bdeb841e28faaa8d
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_pipeline.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_pipeline.c b/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
index e175fa2..9b2d44e 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
@@ -178,7 +178,7 @@ u32 vlv_port_enable(struct intel_pipeline *pipeline,
 
 		for_each_dsi_port(port, intel_dsi->ports) {
 			dsi_port = &disp->port.dsi_port[port];
-
+			vlv_dsi_port_wait_for_fifo_empty(dsi_port);
 			temp = REG_READ(dsi_port->offset);
 			temp &= ~LANE_CONFIGURATION_MASK;
 			temp &= ~DUAL_LINK_MODE_MASK;
-- 
1.7.9.5

